### PR TITLE
Provide single index overview of dev docs

### PR DESCRIPTION
**Pull request by mrnugget**
This only adds the index, since I haven't added redirects yet, which were necessary if we actually remove the folders.

### CODE REVIEW

**Comment by nicksnyder** [comment]
Even if we inline the index on a single page, I think it is still valuable to differentiate between the "how to" links and "background information" links via a heading.

You can watch the video on this page for more background on how we are thinking about docs: https://documentation.divio.com/

**Comment by mrnugget** [comment]
Just to add to what I wrote in Slack: I think I can make the "background information"/"explanation" vs. "how to" divide work, but we need to shift some docs around. I'll try to get that to work on Monday.

**Comment by mrnugget** [comment]
I've updated the PR to only provide the combined overview in index.md

**Review by tsenart — APPROVED** [review]

File: doc/dev/index.md [hunk 1]
```diff
@@ -11,18 +11,121 @@ Sourcegraph development is open source at:
...
+### Other
+
+- [Telemetry](background-information/telemetry.md)
+- [Adding, changing and debugging pings](background-information/adding_ping_data.md)
+
 ## Guidelines
 
-- [Go style guide](https://about.sourcegraph.com/handbook/engineering/languages/go)
-- [TypeScript style guide](https://about.sourcegraph.com/handbook/engineering/languages/typescript)
 - [Code reviews](https://about.sourcegraph.com/handbook/engineering/code_reviews)
```
Comment (pdubroy) [inline doc/dev/index.md:129]:
```suggestion
- [Code reviews](background_information/code_reviews.md)
```

File: doc/dev/index.md [hunk 2]
```diff
@@ -11,18 +11,121 @@ Sourcegraph development is open source at:
...
+- [Terraform](background-information/languages/terraform.md)
+
+#### [Extended guides](background-information/languages/extended_guide/index.md)
+
+- [Terraform Extended Guide](background-information/languages/extended_guide/terraform.md)
+
+### Testing
+
+- [Continuous Integration](background-information/continuous_integration.md)
+- [How to run tests](background-information/testing.md)
```
Comment (pdubroy) [inline doc/dev/index.md:112]: This looks like a how-to :-)

Comment (mrnugget) [reply doc/dev/index.md:112]: Hehe, good point. The title of the document itself is just "Testing" ;)

**Pull request merged**